### PR TITLE
Add The Ability to Randomly Select Reviewers/Assignees From Groups

### PR DESCRIPTION
**Pull request by tommy-meacham**
Hi kentaro-m,
I find myself in the situation where there is a rift in the engineering team I am working in (which I understand is a common problem). The divide in knowledge and practices lies along the separation of our offices.

I therefore looked for a tool that could help us to share the knowledge between offices. The auto_assign bot was perfect for our needs, except that we specifically want to select people from each office, or "group", to review any given PR. The theory is that selecting reviewers from each group will help us bridge the gap in knowledge and bring the two teams closer together.

This PR implements the ability to split collaborators that use this bot into multiple groups, and select reviewers/assignees randomly from each group.

I have designed the implementation to leave all the current functionality intact, and add review/assignee groups on top of your current code.

Let me know what you think!

### CODE REVIEW

File: .github/auto_assign_example.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Set to true to add reviewers to pull requests
```
Comment (kentaro-m) [inline .github/auto_assign_example.yml]: [IMO] There are more setting properties, and it is not clear which property is required, so I sugget to add a comment.

```suggestion
# Set to true to add reviewers to pull requests (required)
```

File: .github/auto_assign_example.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+addReviewers: true
+
+# Set to true to add assignees to pull requests
+addAssignees: true
+
+# A list of reviewers to be added to pull requests (GitHub user name)
```
Comment (kentaro-m) [inline .github/auto_assign_example.yml]:
```suggestion
# A list of reviewers to be added to pull requests (GitHub user name) (required)
```

File: .github/auto_assign_example.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+reviewers:
+  - reviewerA
+  - reviewerB
+  - reviewerC
+
+# A number of reviewers added to the pull request
```
Comment (kentaro-m) [inline .github/auto_assign_example.yml]:
```suggestion
# A number of reviewers added to the pull request (required)
```

File: src/handler.ts [hunk 4]
```diff
@@ -8,55 +8,81 @@ interface AppConfig {
   assignees?: string[],
   numberOfAssignees?: number,
   numberOfReviewers: number,
-  skipKeywords?: string[]
+  skipKeywords?: string[],
+  useReviewGroups: boolean,
```
[the remark this answers is not in the document]
Comment (tommy-meacham) [reply src/handler.ts]: 👍 Added this to the PR.

**Comment by tommy-meacham** [comment]
I liked your suggested changes and added them in to this PR! 😄

**Comment by kentaro-m** [comment]
@tommy-meacham 
Thanks for adding my suggested changes 👍 

After I merged #72 and #78 yesterday, this PR occurs conflicts. I tried to resolve it.
I added commits to resolve conflicts to the following branch, so please added it to this PR if you don't mind. If you have any questions, let me know.

https://github.com/kentaro-m/auto-assign/tree/add-reviewers-group-feature

This commits to resolve conflicts and fix lint errors.
https://github.com/kentaro-m/auto-assign/commit/0ff6901d665c7e2670747a0025c2e1197c6ab37c https://github.com/kentaro-m/auto-assign/commit/bc44851b3e6f3deacee61012a8a06fb1dc5f28bd

**Comment by tommy-meacham** [comment]
Conflicts resolved :)

**Comment by tommy-meacham** [comment]
Hey kentaro-m,
Is there anything I can do to help this get merged?

**Comment by kentaro-m** [comment]
Hi @tommy-meacham.
I have checked your change already. But, this PR has lint errors and build errors so I send the PR for fixing errors to your forked repository https://github.com/tommy-meacham/auto-assign/pull/12.
Can you add the change of https://github.com/tommy-meacham/auto-assign/pull/12 to this PR?

**Comment by tommy-meacham** [comment]
Hey @kentaro-m,
Absolutely!
I didn't see that PR i'm sorry.

**Comment by tommy-meacham** [comment]
Just merged your changes.
Let met me know how else I can help :)

**Pull request merged**